onnxruntime/core/mlas/inc/mlas.h

```diff
@@ -77,6 +77,16 @@ Module Name:
 #define MLAS_SUPPORTS_GEMM_DOUBLE
 #endif
 
+#if (!defined(_MSC_VER)) || (_MSC_VER >= 1930)
+// Visual Studio older than 2022 does not support fp16 intrinsic
+
+#if defined(MLAS_TARGET_ARM64) || defined(MLAS_TARGET_ARM64EC)
+
+#define MLAS_F16VEC_INTRINSICS_SUPPORTED
```

Comment:
add a pure C++ implementation for this case? It can still avoid another pass of main memory.

Reply:
since there is no fp16 data type supports in C++, writing this would be really cobbersome and time consuming.